github-workflows/github/src/calculate_pr_metrics.py

```diff
@@ -0,0 +1,119 @@
+import os
+from github import Github
+from datetime import datetime, timedelta, timezone
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import json
+
+class RepositoryMetrics:
+    def __init__(self, repo_name, time_frame):
+        self.g = Github(os.getenv('GITHUB_TOKEN'))
+        self.repo_name = repo_name
+        self.time_frame = int(time_frame)
+        self.start_date = datetime.utcnow().replace(tzinfo=timezone.utc) - timedelta(days=self.time_frame)
+        self.repo = self.g.get_repo(f"{self.repo_name}")
+

```

Comment:
overall in python I prefer that we use more descriptive parameter names.
```suggestion
        self.github_client = Github(os.getenv('GITHUB_TOKEN'))
        self.repo_name = repo_name
        self.time_frame = int(time_frame)
        self.start_date = datetime.utcnow().replace(tzinfo=timezone.utc) - timedelta(days=self.time_frame)
        self.repo = self.g.get_repo(f"{self.repo_name}")


```